### DISCUSSION TITLE
Reolink Doorbell WiFi 2 Way Audio from HomeKit not working

### DISCUSSION THREAD

**Discussion by dudesuper82** [koush/scrypted#1001] [Q&A] [answered]
Have this in the console when trying to use it with the Reolink plugin works with the onvif plugin

HomeKit]: video/audio detected, discarding further input
[HomeKit]: received first audio packet
[HomeKit]: intercom failed to start RPCResultError: Wrong ONVIF SOAP response
[HomeKit]:     at cb (/onvif/node_modules/onvif/lib/utils.js:74:15)
[HomeKit]:     at o.<anonymous> (/onvif/node_modules/xml2js/lib/parser.js:308:18)
[HomeKit]:     at o.emit (node:events:513:28)
[HomeKit]:     at o.emit (node:domain:489:12)
[HomeKit]:     at o.event [as onerror] (/onvif/node_modules/xml2js/lib/parser.js:123:26)
[HomeKit]:     at emit (/onvif/node_modules/sax/lib/sax.js:624:29)
[HomeKit]:     at error (/onvif/node_modules/sax/lib/sax.js:653:5)
[HomeKit]:     at strictFail (/onvif/node_modules/sax/lib/sax.js:677:7)
[HomeKit]:     at closeTag (/onvif/node_modules/sax/lib/sax.js:871:9)
[HomeKit]:     at o.write (/onvif/node_modules/sax/lib/sax.js:1436:13)
[HomeKit]: @scrypted/reolink:host
[HomeKit]: host:@scrypted/homekit {
[HomeKit]:   cause: undefined
[HomeKit]: }
listen loop connection failed, restarting listener. Wrong ONVIF SOAP response
HomeKit]: ffmpeg exited
[HomeKit]: two way audio failed S [Error]: timeout waiting for client
[HomeKit]:     at Timeout._onTimeout (/@scrypted/homekit/main.nodejs.js:2:1022574)
[HomeKit]:     at listOnTimeout (node:internal/timers:569:17)
[HomeKit]:     at processTimers (node:internal/timers:512:7)
listen loop connection failed, restarting listener. Wrong ONVIF SOAP response

**Comment by mdvz0r** [comment]
hi, got the exact same problem. Any updates on this?

**Reply by kiler129** [reply, marked as answer]
Is your firmware up to date? The one it comes with doesn't properly support 2-way audio.

**Reply by mdvz0r** [reply]
Using firmware v3.0.0.2033_23041302 which seems to be the newest according to the reolink download center.

**Reply by fgonza2** [reply]
same here, using the latest firmware

**Comment by krizzziz** [comment]
Same issue here, have Scrypted setup with the Reolink plugin and have set it up as a doorbell and have Homekit plugin turned on.
Two-way Audio is not working currently, also not working after removing and adding the Reolink doorbellecamera.
I have FW version v3.0.0.2033_23041300

**Comment by GravityRZ** [comment]
EDIT:
i found something important.
it seems that UID is conflicting with homekit.

UID ON: reolink app shows image instantly
homekit talkback does not work

UID OFF: reolink app tries to load image(probably external while it is on local IP) but fails, after that it shows the image
homekit talkback WORKS

this however is not consistent. also with UID OFF it will not work after a while

since homekit uses it's own bridging so it can be acces locally and remote(without setting up a vpn) i suspect that it can not deal with UID.

is there any way to fix this?
normally i do not need UID because i have a vpn but since the app has a hickup i need to leave it on


NOT WORKING
, also on lastest firmware  so clearly this is not the solution for everything

the thing is that it worked when i installed it the first time, after that it stopped working.

can we check the log somehow to see what is going wrong

this is something i see but i can not copy the log from the screen
![image](https://github.com/koush/scrypted/assets/16853535/58d934cd-b840-448f-acd3-6f6cb11ce26f)

**Reply by jericsmith504** [reply]
Same situation except in addition to no two-way audio also no live image.
Scrypted 0.55.0, Reolink plugin 0.0.35 and camera firmware v3.0.0.2033_23041300 was unable to get the camera to work with Homekit (would add, would show a snapshot if I overrode the snapshot URL w/ http://cam-ip/cgi-bin/api.cgi?cmd=Snap&channel=0&rs=1234abc&user=USERNAME&password=PASSWORD)
Loading camera wouldn't work and audio or would get no event notifications either. 

Switched to the ONVIF plugin and used "RuleEngine/MyRuleDetector/Visitor" as the ONVIFDoorbellEventName and it works seemlessly including 2-way audio.

**Reply by fgonza2** [reply]
thanks for the info.  I will try it.  Mine is working again, but the person detection is prob 50% of the time, but the stream is working fine. Any tips on how to improve the detection so homekit records the videos?

**Comment by GravityRZ** [comment]
2 way audio still not working on my end, the rest works perfect.
i only get doorpress notifications in homekit, exactly the way i like it

put stream on 2048Kbits
sensitivity 35
smart detection on 20

i like to see everything what happens in my street.

only false detections when you have ir on at night because bugs get lit up so i have switched of ir

**Reply by bmwcar** [reply]
![123](https://github.com/koush/scrypted/assets/41981098/617a60fa-d657-455f-ba2e-aa65c745e7d5)
https://www.onvif.org/conformant-products/
search your model , and you can found it whether support profile T , only support profile T camera can 2way audio
